tests/ResizerTest.php

```diff
@@ -492,15 +494,88 @@ public function testResizeSameDimensionsRelative()
     }
 
     /**
+     * Tests the resize() method with an event listener.
+     */
+    public function testResizeWithEventListener()
+    {
+        if (!is_dir($this->rootDir)) {
+            mkdir($this->rootDir, 0777, true);
+        }
+
+        (new GdImagine())
+            ->create(new Box(100, 100))
+            ->save($this->rootDir.'/dummy.jpg')
+        ;
+
+        $calculator = $this->getMock('Contao\Image\ResizeCalculatorInterface');
+
+        $calculator
+            ->method('calculate')
+            ->willReturn(new ResizeCoordinates(new Box(100, 100), new Point(0, 0), new Box(100, 100)))
+        ;
+
+        $customImage = $this
+            ->getMockBuilder('Contao\Image\Image')
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
+
+        $eventDispatcher = $this
+            ->getMockBuilder('Symfony\Component\EventDispatcher\EventDispatcher')
+            ->setMethods(['dispatch'])
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
+
+        $eventDispatcher
+            ->method('dispatch')
+            ->willReturnCallback(
+                function ($name, ResizeImageEvent $event) use ($customImage) {
+                    $event->setResizedImage($customImage);
+                }
+            )
+        ;
+
+        $resizer = $this->createResizer($calculator, null, null, $eventDispatcher);
+
+        $image = $this
+            ->getMockBuilder('Contao\Image\Image')
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
+
+        $image
+            ->method('getDimensions')
+            ->willReturn(new ImageDimensions(new Box(200, 200)))
+        ;
+
+        $image
+            ->method('getPath')
+            ->willReturn($this->rootDir.'/dummy.jpg')
+        ;
+
+        $image
+            ->method('getImagine')
+            ->willReturn(new GdImagine())
+        ;
+
+        $configuration = $this->getMock('Contao\Image\ResizeConfigurationInterface');
+        $resizedImage = $resizer->resize($image, $configuration, new ResizeOptions());
+
+        $this->assertEquals($resizedImage, $customImage);
```

Comment:
I think this should be `$this->assertSame($customImage, $resizedImage);`, so that it tests if the two objects are the same instance.